### ISSUE TITLE
Where to put the settings.json for the say command? (and 2 other questions)

### ISSUE THREAD

**Issue by dolphinkite** [jishi/node-sonos-http-api#205]
Hi,

First, is there a forum where I can post various support questions? It doesn't seem right to raise an "issue" for something I don't fully understand or I need help with..?

This being says, here are my issues:
1- I'm trying to get the "say" command to work, but I get the error message from missing_api_key.mp3. However, I did create a file called "settings.json" and I tried to store it in /opt/app, /opt/app/lib/actions and (which I think is the right place if I look at the code) /opt/app/static/tss. That file only contains the following lines:

`{
  "voicerss": "XXX My API key that I got from http://www.voicerss.org/personel/"
}
`

2- Also, I'm trying to implement doorbell scenario. I'm thinking something along the lines of:
curl "http://192.168.x.x:5005/pauseall"
curl "http://192.168.x.x:5005/room2/say/someone%20is%20at%20the%20door"
curl "http://192.168.x.x:5005/resumeall"
Will this work when I have different rooms, some grouped and some not, some playing from local library and others playing from slacker or tunein?

3- How can I queue a specific mp3 file instead of using the "say" command? So I want to pauseall, play specific mp3, resumeall

Thanks for your help.

**Comment by jishi** [comment]
There is no good place today where you can post these questions, except here. In the old sonos forum I had a thread that was active, but since they switched to a new community forum, that got sort of lost in all the noise.

the settings.json file should, according to my understanding of your setup, be in /opt/app, at the same level as server.js. It needs to be named "settings.json" only, if you have created a text file or similar in Windows, it might have a hidden file extension (like `settings.json.txt`).

2- Yes, it would somewhat work. However, the curl command will not wait for the say command to finish, so you would need to add a decent delay yourself today (I will try and fix that in the beta version). Do note that say would use the whole group that room2 belongs to, and there are some music sources that doesn't restore correctly when you invoke say (tunein and pandora, being some) today. Will fix those in beta as well.

3- Right now, you can't do it easily. It's a new feature. Follow this issue for updates: https://github.com/jishi/node-sonos-http-api/issues/136

**Comment by dolphinkite** [comment]
As a side note: I am running this in a docker container. Looking at the folder /opt/app/actions, I only see 22 files, and the one that was added 12 days ago (playlists.js) isn't there. I'm assuming I may be running an outdated version, but I just created the container today... so I'm not sure if this has something to do with it.

**Comment by jishi** [comment]
Ah, if you are running docker, it's a bit different story. Then you need to "link" an external file into your docker containers file-system. Also, the pre-built docker image is a bit outdated as well. If you don't have an explicit need for docker, I would discourage it right now since it also requires specialized startup and disabling of NAT to function correctly.

**Comment by dolphinkite** [comment]
Thanks for your answer. I do need to run it in a container because I'm running it on a Synology NAS. However, my understanding of Docker is reasonable and I'm successfully running a few containers. I have no problem using the --net=host command to make sure the container has access to the network (and it does, the other functions are working).

Specifically regarding the sonos-http, I did notice that the docker was outdated; the old Dockerfile would call "dockerstart.sh" which in return would do a "git clone" of the latest files, so it should work. However, it seems the latest version isn't calling that file anymore so I'm not sure why dockerstart.sh is still in github. Anyway, I'll pull the latest code and build my own container - that shouldn't be too hard.

This being said, for the settings.json, I was able to create it directly in the container by attaching to the specific docker container using the `docker exec -ti containerName bash` command and then going to the /opt/... folder and create that file locally using VI. 

I managed to get further by putting settings.json in the /opt/app folder (the same as server.js). My problem (for others who may encounter this) is that I needed to stop and start the container for the new file to be "visible" to the server.

Fo your info, the generated mp3 was unreadable. Sonos told me "Unable to play xxx.mp3 - the song is not encoded correctly". After running it through mp3diags, I realised it was a txt file that contained the sentence: ""ERROR: The language does not support!". It seems I MUST pass a language parameter for this to work.

Thanks for your help. It now works. I am looking forward to the beta version.
